cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -130,6 +130,9 @@ struct ARROW_EXPORT S3Options {
   /// Whether OutputStream writes will be issued in the background, without blocking.
   bool background_writes = true;
 
+  /// Whether to allow creation of new buckets
+  bool allow_create_buckets = false;
```

Comment:
Could you also update `S3Options::FromUri()` to support this?

Reply:
Good idea. I've added that.

Reply:
Let's make it two options for flexibility (creation and deletion).

Also wrt. naming, should this be `allow_bucket_creation`? @lidavidm

Reply:
Perhaps `create_buckets` and `delete_buckets`? That reads more consistently with `background_writes` to me. Or `allow_creating_buckets`

Reply:
I liked `create_buckets` but since there is a setter method and I want it named the same thing as the property (for clear error hints), I felt like `allow_bucket_creation` was the better choice.